teemaderegister-react/src/components/TopicTable.jsx

```diff
@@ -0,0 +1,112 @@
+import React, {useEffect, useState} from 'react'
+import {connect} from 'react-redux'
+import {getSupervisorTopics} from '../actions/SupervisorActions'
+import { DownloadOutlined } from '@ant-design/icons'
+import {setDocTitle} from '../utils/Helpers'
+import {Button, Table, Tag} from 'antd'
+
+const defaultColumns = [
+    {
+        title: 'Title',
+        dataIndex: 'title',
+        key: 'title'
+    },
+    {
+        title: 'Author',
+        dataIndex: 'author',
+        render: data => {
+            return (`${data.firstName} ${data.lastName}`)
+        }
+    },
+    {
+        title: 'Types',
+        dataIndex: 'types',
+        render: data => {
+            return (data.map((value) => {
+                // eslint-disable-next-line react/jsx-key
+                return <Tag color='blue'>{value}</Tag>
+            }))
+        }
+    },
+    {
+        title: 'Registered',
+        dataIndex: 'registered',
+        render: data => {
+            return (data.split('T')[0])
+        }
+    }
+]
```

Comment:
would define default ones in one array,e.g
```
const defaultColumns = [
    {
        title: 'Title',
        dataIndex: 'title',
        key: 'title'
    },
    {
        title: 'Author',
        dataIndex: 'author',
        render: data => {
            return (`${data.firstName} ${data.lastName}`)
        }
    },
    {
        title: 'Types',
        dataIndex: 'types',
        render: data => {
            return (data.map((value) => {
                // eslint-disable-next-line react/jsx-key
                return <Tag color='blue'>{value}</Tag>
            }))
        }
    }
]
```
and do separately registered and defended
like 
```
const defendedColumns = [
    ...defaultColumns,
    {
        title: 'Defended',
        dataIndex: 'defended',
        key: 'defended',
        render: data => {
            return data.split('T')[0]
        }
    },
    {
        title: 'File',
        dataIndex: 'file',
        key: 'file',
        render: link => {
            return (
                <Button type='primary' icon={<DownloadOutlined />} href={link} />)
        }
    }
]
```

Reply:
so there will be less repetition of code